pmdarima/arima/_auto_solvers.py

```diff
@@ -183,6 +192,18 @@ def step_through(self):
                                            self.p < self.max_p, q=self.q + 1,
                                            p=self.p + 1)
 
+            # break loop if execution time exceeds the timeout threshold
+            dur = (datetime.now() - start_time).total_seconds()
+            if self.max_dur and dur > self.max_dur:
+                warnings.warn('early termination of stepwise search due to '
+                              'max_dur threshold')
+                break
+
+        # check if the search has ended after max_k tries
+        if self.k > self.max_k:
+            warnings.warn('stepwise search has reached the maximum number '
+                          'of tries to find the best fit model')
```

Comment:
Should this `break`?

Reply:
at this point, the control is already outside the ``while`` loop. The warning helps us to assert that stepwise ran for max_steps times.

Reply:
I'll be making a change to raise this warning only when ``max_steps`` threshold was supplied through the ``StepwiseContext`` and ``self.k > max_steps``. Otherwise, this warning always pops up!